6653 contenttype menu (#8032)
plone/volto: docs/configuration/configuration-registry.md

Reviewer: @sneridagh and @stevepiercy  I would like your opinion on docs organization, I am not very good at that. This file has a name that's very broad and explains a very small piece of the whole config registry. Do you think it's ok for this PR?

Reviewer 2: See my earlier suggestion at https://github.com/plone/volto/pull/8032#pullrequestreview-3988667571. This reads like a how to guide.

Reviewer 3: I agree, we might want to move it there. However, since configuration (customization) is such a big topic, maybe deserve its own section. I think like @pnicolli , I'd rely on your judgement @stevepiercy

Reviewer 2: As they say, "Size doesn't matter, but how you use it." The same applies to documentation.

As I've read through the configuration docs, I can see how I would reorganize them, following Diátaxis, but that's for a later PR. Step 1: get content. Step 2: organize content. Step 3: profit!

Specifically, I see a Configuration subsection under How-to guides, including "Extend Vite", this file, "Component registry", "Plate code block languages", and so on.

Reviewer 3: oh, forget my comment, is already done.

> d---

Reviewer 2: 💣 

https://volto--8032.org.readthedocs.build/8032/configuration/configuration-registry.html

```suggestion
---
```

Author: Oops. I removed it

myst:
  html_meta:

>     "description": "How to access and modify the configuration regisrey"

Reviewer 2: ```suggestion
    "description": "How to access and modify the configuration registry"
```

    "property=og:description": "How to access and modify the configuration registry"
    "property=og:title": "Customize Seven using the configuration registry"
    "keywords": "Seven, @plone/registry, registry, configuration"
---

# Customize Seven using the configuration registry

The registry is the central place for configuring your Seven application.

> Configuration is typically located either in the index.ts file at the add-ons root level or in config/settings.ts.

Reviewer 2: ```suggestion
Configuration is typically located either in the {file}`index.ts` file at the add-on's root level or in {file}`config/settings.ts`.
```

Author: I changed this sentence a little, since it's wrong. It's not either/or, but rather the main config file is (usually? or always?) the `index.ts` at root level, and it is optionally divided into smaller chunks, which are placed in the `config` directory, like `@plone/layout` does.

> Configuration is typically located in the {file}`index.ts` file at the add-on's root level and inside the `config` directory, if it exists.

Reviewer 2: `{file}` is used for directories, too.

```suggestion
Configuration is typically located in the {file}`index.ts` file at the add-on's root level and inside the {file}`config` directory, if it exists.
```

> In this file, you can modify existing entries or add new ones as needed.

Reviewer 2: ```suggestion
In this file, you can modify existing entries or add new ones.
```

> In this file, you can modify existing entries or add new ones.

Reviewer 2: Instead of messing with the issue of plurality, I avoid it by referring to the example single file instead.

```suggestion
The following configuration file example shows how to modify existing entries or add new ones.
```

```ts
import type { ConfigType } from '@plone/registry';
import MyBlockInfo from 'my/addon/blocks/MyBlockInfo';

export default function install(config: ConfigType) {

>   // Example: registering a new block

Reviewer 2: ```suggestion
  // Register a new block
```

  config.blocks.blocksConfig.myBlock = MyBlockInfo;
  
  // Example: Adding a content type to the "most used" section

>   //          of the content type menu in the toolbar 

Reviewer 2: ```suggestion
  // Add a content type to the "most used" section
  // of the content type menu in the toolbar
```
